### DISCUSSION TITLE
How do I use Pyinstaller with an external config file?

### DISCUSSION THREAD

**Discussion by innosilion** [pyinstaller/pyinstaller#8600] [Help]
I'm using PyInstaller on Linux to make an EXE file of my project.

I would like to use the --onefile option to have a clean result and an easy-to-distribute file/program.

My program uses a exe.config file for storing configuration options. This file could be customized by users.

How can i do?

**Comment by rokm** [comment]
Where do you plan to put that config file? If next to executable, you can refer to it relative to the `sys.executable`. Or you can use something like [appdirs](https://pypi.org/project/appdirs/) to read config file from standard location in user's home directory.

**Reply by innosilion** [reply]
I want to dynamic to load config file , and those code is integrated into executable programs packaged in pyinstaller.
And the config file and executable program are in the same directory.

But it is wrong  when we use  sys.executable to find the config file . 
The path to sys. executable is  /usr/bin/Python.

How can i find the right path to load.

**Reply by rokm** [reply]
When you run your script in python interpreter, `sys.executable` points to your point interpreter. In PyInstaller-frozen application, `sys.executable` points to the executable.

**Reply by innosilion** [reply]
I try use it again.
It shows :
failed: module 'sys' has no attribute 'executable'.

what is reason for this error?

**Reply by rokm** [reply]
What happens if you freeze the following program:
```python
import sys
print(sys.executable)
```

**Reply by innosilion** [reply]
sorry, I just typed the code incorrectly.
but it shows when using:
import sys
print(sys.executable)

$ sys.executable ----> /usr/bin/Python

**Reply by rokm** [reply]
> sorry, I just typed the code incorrectly. but it shows when using: import sys print(sys.executable)
> 
> $ sys.executable ----> /usr/bin/Python

Even after freezing it with PyInstaller?

**Reply by innosilion** [reply]
sys. excutable is not the path to an executable program after freezed.

**Reply by innosilion** [reply]
yes.

**Reply by rokm** [reply]
On what OS and what PyInstaller version?

**Reply by innosilion** [reply]
linux OS and  6.8.0 version of  PyInstaller.

**Reply by rokm** [reply]
And this is the example program I gave you earlier, or your program?

**Reply by innosilion** [reply]
In my program.

**Reply by rokm** [reply]
Can you try with my example program? 

Because if `sys.executable` is pointing to python interpreter, my guess would be that you are spawning a subprocess using system interpreter that runs the script in question...

**Reply by innosilion** [reply]
ok, i try it.
The result is showing:
[GCC 8.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> print(sys.executable)
/usr/bin/python3

**Reply by rokm** [reply]
That's not PyInstaller-frozen application.

**Reply by innosilion** [reply]
You are right. 
I just use subprocess to run  print(sys.executable)  .
So, how can i do in the subprocess program.

**Reply by innosilion** [reply]
Why does that no work in subprocess?
Have any workaround for it?

**Reply by rokm** [reply]
If you are spawning subprocesses that need python interpreter, then there is no point in using PyInstaller in the first place.

**Reply by innosilion** [reply]
okey, i got it.
